src/layouts/Settings/PrivacySettings.tsx

```diff
@@ -0,0 +1,181 @@
+  PopoverContent,
+  PopoverArrow,
+  PopoverBody,
+  Text,
+} from "@chakra-ui/react"
+import {
+  FormErrorMessage,
+  FormLabel,
+  IconButton,
+} from "@opengovsg/design-system-react"
+import { FormTitle } from "components/Form"
+import { useEffect, useState } from "react"
+import { useFormContext, useFormState } from "react-hook-form"
+import { BiCopy } from "react-icons/bi"
+
+import { Section, SectionHeader } from "layouts/components"
+
+import { generatePassword } from "utils/password"
+import { PASSWORD_REGEX } from "utils/validators"
+
+import { SitePasswordSettings } from "types/settings"
+
+import { FormToggle } from "./components/FormToggle"
+
+interface PrivacySettingsProp {
+  isError: boolean
+}
+
+export const PrivacySettings = ({
+  isError,
+}: PrivacySettingsProp): JSX.Element => {
+  const { register, trigger, watch, setValue, getValues } = useFormContext()
+  const { errors } = useFormState<SitePasswordSettings>()
+  const { onCopy, hasCopied } = useClipboard(watch("password"))
+  const [isBuffering, setIsBuffering] = useState(false)
+
+  const isPrivatised = watch("isStagingPrivatised")
+
+  const generateAcceptablePassword = () => {
+    const regex = new RegExp(PASSWORD_REGEX)
+    let password = ""
+    while (!regex.test(password)) {
+      password = generatePassword()
+    }
+    return password
+  }
+
+  useEffect(() => {
+    // Generate a default password on toggle if no password exists
+    if (!getValues("password")) {
+      const newPassword = generateAcceptablePassword()
+      setValue("password", newPassword)
+      trigger("password")
+    }
+  }, [isPrivatised])
+
+  return (
+    <Section id="privacy-fields">
+      <SectionHeader label="Privacy Settings" />
+      <VStack spacing="1.5rem" align="flex-start" w="50%">
+        <FormControl isDisabled={isError}>
+          <Flex justifyContent="space-between" w="100%">
+            <Box>
+              <FormTitle>Privatise staging site</FormTitle>
+              <FormLabel.Description color="text.description">
+                Password protect access to your staging site without affecting
+                your live site.
+              </FormLabel.Description>
+            </Box>
+
+            {/* TODO: Swap out component with Checkbox, along with other toggle components */}
+            <FormToggle name="isStagingPrivatised" />
+          </Flex>
+        </FormControl>
+        {watch("isStagingPrivatised") && (
+          <VStack spacing="0.75rem" align="flex-start" w="100%">
+            <Box>
+              <FormTitle>Username</FormTitle>
+              <FormLabel.Description color="text.description">
+                This cannot be changed.
+              </FormLabel.Description>
+            </Box>
+            <Input isDisabled placeholder="user" />
+            <Box>
+              <FormTitle>Password</FormTitle>
+              <FormLabel.Description color="text.description">
+                You should change this every 90 days
```

Comment:
nit: shouldnt this have a fullstop similar to other labels?

Reply:
Oh it should, https://github.com/isomerpages/isomercms-frontend/pull/1316/commits/1f72a35279945dabf24db1903ee56ea778260075